### PR TITLE
NXP-26733: Missing nuxeo-platform-login-shibboleth-jsf-9.10*.jar

### PR DESCRIPTION
**Pull request by RSalem07**

### CODE REVIEW

File: marketplace/pom.xml [hunk 1]
```diff
@@ -1,41 +1,59 @@
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
-  <modelVersion>4.0.0</modelVersion>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
```
Comment (troger) [inline marketplace/pom.xml]: We use and indentation of 2 spaces for XML files, see https://doc.nuxeo.com/corg/xml-or-xml-like-files-code-style/.

File: marketplace/pom.xml [hunk 2]
```diff
@@ -1,41 +1,59 @@
...
+            <groupId>org.nuxeo.ecm.platform</groupId>
+            <artifactId>nuxeo-platform-login-shibboleth</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.nuxeo.ecm.platform</groupId>
+            <artifactId>nuxeo-platform-login-shibboleth-jsf</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.nuxeo.ecm.platform</groupId>
+            <artifactId>nuxeo-usermapper</artifactId>
```
Comment (troger) [inline marketplace/pom.xml]: You can remove that dependency, it is already a dependency of `org.nuxeo.ecm.platform:nuxeo-platform-login-shibboleth` so it will end up in the MP.

File: marketplace/src/main/resources/package.xml [hunk 3]
```diff
@@ -1,21 +1,27 @@
...
+    <description>The Shibboleth Authentication addon allows to let your users authenticate against Shibboleth compliant
+        identity management federation systems, providing a Single Sign On experience with other applications also
+        plugged to the same system.
+    </description>
+    <vendor>Nuxeo</vendor>
+    <classifier>Open Source</classifier>
+    <home-page>http://doc.nuxeo.com/display/NXDOC/Shibboleth+Authentication
+    </home-page>
+    <platforms>
+        <platform>server-11.1-SNAPSHOT</platform>
```
Comment (troger) [inline marketplace/src/main/resources/package.xml]: You must keep `@DISTRIB_VERSION@` and not fix it to a version, why did you change it?

**Review by troger — CHANGES_REQUESTED** [review]
Commit message should not mention `9.10` as you are working on the master branch.
> NXP-26733: missing nuxeo-platform-login-shibboleth-jsf jar
or
> NXP-26733: handle JSF UI bundles

Update your IDE to follow the XML formatting (2 spaces).

Seems that `nuxeo-usermapper` jar is not part of the MP (when building locally), need to find why.

File: marketplace/src/main/resources/package.xml (hunk 3 above)
Comment (RSalem07) [reply marketplace/src/main/resources/package.xml]: Yes, I agree, it was a mistake.

File: marketplace/pom.xml (hunk 2 above)
Comment (RSalem07) [reply marketplace/pom.xml]: I agree and now the use-mapper dependency is end up in the MP

**Comment by RSalem07** [comment]
The following points are fixed, Please review 

- Commit message should not mention `9.10` as you are working on the master branch. => **Done** I used the **NXP-26733: missing nuxeo-platform-login-shibboleth-jsf jar** comment message.
- You can remove that dependency, it is already a dependency of `org.nuxeo.ecm.platform:nuxeo-platform-login-shibboleth` so it will end up in the MP. => **Done**
- Seems that `nuxeo-usermapper` jar is not part of the MP (when building locally), need to find why. => **Done**, now the jar is available in MP
- We use and indentation of 2 spaces for XML files, see https://doc.nuxeo.com/corg/xml-or-xml-like-files-code-style/. => **Done**, use 2 spaces instead of a tab character.
- You must keep `@DISTRIB_VERSION@` and not fix it to a version, why did you change it? => **Done**.

File: marketplace/src/main/assemble/assembly.xml [hunk 4]
```diff
@@ -1,31 +1,190 @@
 <project name="shibboleth-assembly" default="build" xmlns:nx="urn:nuxeo-build" xmlns:artifact="urn:nuxeo-artifact">
-  <taskdef resource="org/nuxeo/build/antlib.xml" uri="urn:nuxeo-build" />
-  <taskdef resource="org/nuxeo/build/artifact/antlib.xml" uri="urn:nuxeo-artifact" />
+  <taskdef resource="org/nuxeo/build/antlib.xml" uri="urn:nuxeo-build"/>
```
Comment (troger) [inline marketplace/src/main/assemble/assembly.xml]: Missing space at the end before `/>` in the whole file, IDE configuration I think.

File: marketplace/src/main/resources/install.xml [hunk 5]
```diff
@@ -1,5 +1,6 @@
+<!-- See documentation at http://doc.nuxeo.com/x/IgIz -->
 <install>
-  <update file="${package.root}/install/bundles" todir="${env.bundles}" />
-  <copy dir="${package.root}/templates" todir="${env.templates}" overwrite="true" />
-  <config addtemplate="shibboleth" />
+  @BUNDLES@
+  @LIBS@
+  @PROPERTIES@
```
Comment (troger) [inline marketplace/src/main/resources/install.xml:5]: Missing `@TEMPLATES@`

File: marketplace/src/main/assemble/assembly.xml [hunk 6]
```diff
@@ -1,31 +1,190 @@
...
+  &lt;copy file=&quot;${package.root}/install/artifacts-shibboleth.properties&quot;${line.separator}
+    todir=&quot;${env.server.home}/sdk&quot; overwrite=&quot;true&quot; /&gt;${line.separator}
+  &lt;copy file=&quot;${package.root}/install/test-artifacts-shibboleth.properties&quot;${line.separator}
+    todir=&quot;${env.server.home}/sdk&quot; overwrite=&quot;true&quot; /&gt;"/>
+
+    <copy file="src/main/resources/install.xml" todir="${outdir}/marketplace" overwrite="true">
+      <filterset>
+        <filter token="BUNDLES" value="${install.bundles}"/>
+        <filter token="LIBS" value="${install.libs}"/>
+        <filter token="PROPERTIES" value="${install.properties}"/>
```
Comment (troger) [inline marketplace/src/main/assemble/assembly.xml]: Missing `<filter token="TEMPLATES" value="${install.templates}" />`.

File: marketplace/src/main/assemble/assembly.xml [hunk 7]
```diff
@@ -1,31 +1,190 @@
...
+      </then>
+    </if>
+
+    <var name="install.properties"
+         value="${line.separator}
+  &lt;copy file=&quot;${package.root}/install/artifacts-shibboleth.properties&quot;${line.separator}
+    todir=&quot;${env.server.home}/sdk&quot; overwrite=&quot;true&quot; /&gt;${line.separator}
+  &lt;copy file=&quot;${package.root}/install/test-artifacts-shibboleth.properties&quot;${line.separator}
+    todir=&quot;${env.server.home}/sdk&quot; overwrite=&quot;true&quot; /&gt;"/>
+
```
Comment (troger) [inline marketplace/src/main/assemble/assembly.xml:171]: The Shibboleth MP has a `shibboleth` template that you can find in `src/main/resources/templates/`.
We need to add it to the `install.xml` file so that it will be deployed on the server when installing the Shibboleth MP.

```
  <var name="install.templates" value="" />
    <if>
      <available file="${outdir}/marketplace/install/templates" />
      <then>
        <var name="install.templates"
             value="${line.separator}
  &lt;copy dir=&quot;${package.root}/install/templates&quot; todir=&quot;${env.templates}&quot; overwrite=&quot;true&quot; /&gt;${line.separator}
  &lt;config addtemplate=&quot;shibboleth&quot; /&gt;${line.separator}" />
      </then>
    </if>
```

For instance, the Drive MP does the same thing
https://github.com/nuxeo/marketplace-drive/blob/master/marketplace/src/main/assemble/assembly.xml
https://github.com/nuxeo/marketplace-drive/blob/master/marketplace/src/main/assemble/assembly.xml#L180
https://github.com/nuxeo/marketplace-drive/blob/master/marketplace/src/main/assemble/assembly.xml#L196

File: marketplace/src/main/resources/package.xml [hunk 8]
```diff
@@ -1,7 +1,10 @@
 <?xml version="1.0"?>
-<package type="addon" name="shibboleth-authentication" version="@VERSION@">
+<package type="addon" name="shibboleth-authentication" version="2.7.0-SNAPSHOT">
```
Comment (troger) [inline marketplace/src/main/resources/package.xml]: Keep `@VERSION@`.

**Review by troger — CHANGES_REQUESTED** [review]
Added new comments.

You have removed the `<config addtemplate="shibboleth" />` from the `install.xml` so we lost the `templates` part, I've added comments to explain how to add it. 
The Drive MP would have been a better sample as it handles Core and JSF UI bundles + templates (instead of CSV MP that does not have the `templates` part...).

https://github.com/nuxeo/marketplace-drive/blob/master/marketplace
https://github.com/nuxeo/marketplace-drive/blob/master/marketplace/src/main/assemble/assembly.xml

**Comment by RSalem07** [comment]
- **The follow points are fixed**  Please feel free to review
- Project structure adapted to add Template (in Csv as mentioned there is no Template configuration) to be like the marketplace Drive project; add the install/template folder under resources  and i puted the config xml under nxserver directory. 
- Keep `@VERSION@`. => **Done**
- Missing space at the end before `/>` in the whole file, IDE configuration I think. => **Done**

**Review by troger — APPROVED** [review]

**Comment by troger** [comment]
:+1:

Squash all commits in one before doing the "Rebase and merge" action.

**Comment by ataillefer** [comment]
:+1:
